src/kernels/jupyter/launcher/jupyterNotebookProvider.ts

```diff
@@ -51,7 +52,7 @@ export class JupyterNotebookProvider implements IJupyterNotebookProvider {
             token: options.token,
             localJupyter: isLocalConnection(options.kernelConnection)
         });
-
+        Cancellation.throwIfCanceled(options.token);
```

Comment:
Created helper function to check if session is invalid or if token was cancelled.